sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -75,9 +76,22 @@ private[sql] class HiveSessionCatalog(
   // | Methods and fields for interacting with HiveMetastoreCatalog |
   // ----------------------------------------------------------------
 
-  override def getDefaultDBPath(db: String): String = {
-    val defaultPath = hiveconf.getVar(HiveConf.ConfVars.METASTOREWAREHOUSE)
-    new Path(new Path(defaultPath), db + ".db").toString
+  // This function is to get the path for creating a non-default database
+  override def createDatabasePath(dbName: String, path: Option[String] = None): String = {
```

Comment:
we shouldn't use the hive confs here. just use the sqlconf and create a default warehouse folder config option. 

If anything, we should just pre-populate the default warehouse folder option with the value loaded from hive-site.